pypdf/_writer.py

```diff
@@ -3112,13 +3112,8 @@ def _create_outline_item(
     if color:
         if isinstance(color, str):
             color = hex_to_rgb(color)
-        prec = decimal.Decimal("1.00000")
         outline_item.update(
-            {
-                NameObject("/C"): ArrayObject(
-                    [FloatObject(decimal.Decimal(c).quantize(prec)) for c in color]
-                )
-            }
+            {NameObject("/C"): ArrayObject([FloatObject(c) for c in color])}
```

Comment:
This looks actually wrong to me... shouldn't it be 

See "Table 164 – Entries common to all annotation dictionaries":

(Optional; PDF 1.1) An array of numbers in the range 0.0 to 1.0,
representing a colour used for the following purposes:
The background of the annotation’s icon when closed
The title bar of the annotation’s pop-up window
The border of a link annotation
The number of array elements determines the colour space in which the
colour shall be defined:
0 No colour; transparent
1 DeviceGray
3 DeviceRGB
4  DeviceCMYK

Reply:
Meaning if this is a hex-color, something weird is happening

Reply:
My understanding
I agree it defines a color : it is either an array of: 
  * 1 float = gray color from Black(0) to White(0)
  * 3 floats = 3 floats define R/G/B not not on scale 0-255 each but 0.0-1.0
  * 4 floats = 4 floats for CMYK

depending on type of annots , this can be either : the color of background, pop-up,...